snow/consensus/snowball/consensus.go

```diff
@@ -40,15 +40,19 @@ type Consensus interface {
 	Finalized() bool
 }
 
-// NnarySnowball augments NnarySnowflake with a counter that tracks the total
-// number of positive responses from a network sample.
-type NnarySnowball interface{ NnarySnowflake }
-
-// NnarySnowflake is a snowflake instance deciding between an unbounded number
-// of values. After performing a network sample of k nodes, if you have alpha
-// votes for one of the choices, you should vote for that choice. Otherwise, you
-// should reset.
-type NnarySnowflake interface {
+// ConsensusFactory produces NnarySnowflake instances
+type ConsensusFactory interface {
+	New(params Parameters, choice ids.ID) NnarySnow
+	NewUnary(params Parameters) UnarySnow
```

Comment:
Can we just have a single `New` function and expect the parameters to be passed into the the implementation so that we can have a single function exposed?

Reply:
These return different interfaces